Documentation/architecture.rst

```diff
@@ -394,13 +461,26 @@ are routable outside of the container cluster, communication with external
 hosts requires little more than enabling L3 forwarding on each of the Linux
 nodes.
 
-External Connectivity with Overlay Routing
-==========================================
+External Network Connectivity
+=============================
+
+If the destination of a packet relies outside of the cluster. Cilium will
+delegate routing to the routing subsystem of the cluster node to use the
+default route which is installed on the node of the cluster.
+
+As the IP addresses used for the **cluster prefix** are typically allocated
+from RFC1918 private address blocks and not publicly routeable. Cilium will
+automatically masquerade the source IP address of all traffic that is leaving
+the cluster. This behaviour can be disabled by running ``cililum-agent`` with
+the option ``--masquerade=false``.
 
-However, in the case of ''Overlay Routing'', accessing external hosts requires
-additional configuration.
+Public Endpoint Exposure
+========================
 
-In the case of containers accepting inbound connections, such services are
+In direct routing mode, endpoint IPs can be publicly routeable IPs and no
+additional action needs to be taken.
+
+In overlay mode, endpoints accepting inbound connections, such services are
```

Comment:
In overlay mode, endpoints that are accepting inbound connections are likely exposed via some kind of load-balancing layer...